src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,184 @@
+/* stylelint-disable selector-class-pattern */
+import { useEffect, useState } from 'react';
+import Calendar from 'react-calendar';
+import 'react-calendar/dist/Calendar.css'; // css import
+import styled from 'styled-components';
+import { IcArrowBottomWhite, IcArrowTopWhite, IcCalendar } from '../../assets';
+import CustomCalendar from './CustomCalendar';
+
+type ValuePiece = Date | null;
+
+type SelectedDate = ValuePiece | [ValuePiece, ValuePiece];
+
+const CommonCalendar = () => {
+  const today = new Date();
+  const year = today.getFullYear();
+  const [selectedDate, setSelectedDate] = useState<SelectedDate>(today);
+  const [clickedYear, setClickedYear] = useState(year);
+  const [clickedMonth, setClickedMonth] = useState(today.getMonth() + 1);
+
+  const [isCalendarClicked, setIsCalendarClicked] = useState(false);
+
+  const customWeekdays = ['S', 'S', 'M', 'T', 'W', 'T', 'F'];
+
+  const handleClickCalendar = () => {
+    setIsCalendarClicked((prev) => !prev);
+  };
+
+  const handleClickPrevBtn = () => {
+    setClickedYear((prevYear) => prevYear - 1);
+  };
+
+  const handleClickNextBtn = () => {
+    setClickedYear((prevYear) => prevYear + 1);
+  };
+
+  const handleClickMonth = (month: number) => {
+    const newDate = new Date(clickedYear, month - 1, today.getDate());
+    setClickedMonth(month);
+    setSelectedDate(newDate);
+    setIsCalendarClicked(false);
+  };
+
+  useEffect(() => {
+    console.log(selectedDate);
+  });
+
+  return (
+    <CalendarContainer>
+      <NavContainer>
+        <IcCalendar onClick={handleClickCalendar} />
+        <DateContainer>
+          <Year>{clickedYear}년</Year>
+          <Month>{clickedMonth}월</Month>
+        </DateContainer>
+        {isCalendarClicked ? (
+          <CustomCalendarContainer>
+            <IcArrowTopWhite onClick={handleClickCalendar} />
+            <CustomCalendar
+              date={{ clickedYear, clickedMonth }}
+              unsolvedMonths={[]} // 이 배열에 비활성화하고 싶은 월을 추가합니다.
+              handleClickPrevBtn={handleClickPrevBtn}
+              handleClickMonth={handleClickMonth}
+              handleClickNextBtn={handleClickNextBtn}
+            />
+          </CustomCalendarContainer>
+        ) : (
+          <IcArrowBottomWhite onClick={handleClickCalendar} />
+        )}
+      </NavContainer>
+      <StyledCalendar
+        onChange={setSelectedDate}
```

Comment:
P1: 하위 컴포넌트에서는 상위 컴포넌트에서 데이터를 어떻게 저장하는지 알 필요가 없기 때문에, 특별한 이유가 없는 한 상위 컴포넌트의 값을 사용할 때는 아래와 같이 컴포넌트 인터페이스를 설계해주세요 ! (절대 setXXX 프롭을 받지 마세요)
```suggestion
        onChange={(value) => setSelectedDate(value)}
```

Reply:
ASK: 그런데 이 함수를 사용하는 부분이 딱히 보이지 않는데요, StyledCalendar 컴포넌트는 어디있나요 ??

Reply:
혹시 이 부분 이해가 잘 안되는데 .. 설명 한번 더 가능한가요 ?  StyleCalendar 는 react-calendar 에 스타일을 주기 위해 사용한 스타일 컴포는트 입니다.